### PR TITLE
Avoiding puppet returning 2 instead of 0

### PR DESCRIPTION
**Pull request by magmax**
This small change avoids puppet to treturn 2 instead of 0 when there are no changes to be done.

### CODE REVIEW

File: lib/puppet/provider/package/chocolatey.rb [hunk 1]
```diff
@@ -178,7 +178,7 @@ def self.instances
           else
             values = line.split(' ')
           end
-          packages << new({ :name => values[0], :ensure => values[1], :provider => self.name })
+          packages << new({ :name => values[0].downcase, :ensure => values[1], :provider => self.name })
```
Comment (ferventcoder) [inline lib/puppet/provider/package/chocolatey.rb]: Hi @magmax - Your commit message doesn't say much, so I'm going to ask, why is this necessary? What version where you on before? 

Was it 0.5.0 / 0.5.1? https://github.com/chocolatey/puppet-chocolatey/commit/cb7f0e5e33ddef445af4a03b9ab339588a9dc698

Comment (magmax) [inline lib/puppet/provider/package/chocolatey.rb]: Oh, I`m sorry!

I was using version 0.5.0, trying to install nssm with this module:

```
class nssm (
	$version = latest,
	$source  = 'SEE NOTES',
) inherits nssm::params {

	package{'nssm':
		ensure   => $version,
		source   => $source,
	}

}
```

where the source was a miror of nuget central. The package is called "NSSM" and puppet returned 2 again and again. After this change it started to work.

I already know it is being done in the caller, but I do not know if this method is overriding an API method that can be called in other place.

Sadly, I cannot reproduce it right now; my whole environment is failing right now with other problem.

Comment (ferventcoder) [inline lib/puppet/provider/package/chocolatey.rb]: You want to be on at least 0.5.1 - 0.5.0 has issues on Linux masters.
